### ISSUE TITLE
DNN Unknown layer type Mean

### ISSUE THREAD

**Issue by Jaykob** [opencv/opencv#9519]
- OpenCV => 3.3.0 (current master)
- Operating System / Platform => macOS 10.12.6
- Compiler => clang

##### Detailed description

I'm trying to load a relatively simple tensorflow model which uses batch norm with the dnn module. and I get the following error.
I also applied PR #9517 but that didn't change anything.
Is there something I'm missing or any hint what I could do to successfully import that model?

```
OpenCV Error: Unspecified error (Unknown layer type Mean in op conv1/conv1/moments/Mean) in populateNet, file /Users/jaykob/opencv/modules/dnn/src/tensorflow/tf_importer.cpp, line 960
/Users/jaykob/opencv/modules/dnn/src/tensorflow/tf_importer.cpp:960: error: (-2) Unknown layer type Mean in op conv1/conv1/moments/Mean in function populateNet
```

**Comment by dkurt** [comment]
@Jaykob, There are at least 4 batch normalization layers. Please provide a minimal working example or just a certain version of layer.
`tf.nn.batch_normalization`
`tf.layers.batch_normalization`
`tf.contrib.keras.backend.batch_normalization`
`tf.contrib.layers.batch_norm`

**Comment by Jaykob** [comment]
We use `tf.contrib.layers.batch_norm`

**Comment by dkurt** [comment]
@Jaykob, the general way to prepare a network for import is a call of `freeze_graph.py` and `optimize_for_inference.py` tools. Special cases might require manual graph transformations by `transform_graph` tool. Please try to follow one of the ways proposed at https://github.com/opencv/opencv/pull/9517 or https://github.com/opencv/opencv/issues/9459#issuecomment-326287052.
